crates/biome_js_analyze/src/lint/nursery/use_index_of/find_index_comparable_expression.rs

```diff
@@ -0,0 +1,78 @@
+use biome_js_syntax::{
+    AnyJsExpression, AnyJsFunctionBody, JsAssignmentExpression, JsBinaryExpression,
+    JsLogicalExpression, JsReturnStatement, JsSyntaxNode, JsVariableDeclaration,
+};
+use biome_rowan::AstNode;
+
+fn node_is_undefined(node: &AnyJsExpression) -> bool {
+    node.syntax().text_trimmed() == "undefined"
+}
+
+fn extract_simple_compare_match(
+    expression: &JsBinaryExpression,
+    parameter_name: &String,
+) -> Option<JsSyntaxNode> {
+    if expression.operator_token().ok()?.text_trimmed() != "===" {
```

Comment:
I think it is cleaner to compare the token kind:

```suggestion
    if expression.operator_token().ok()?.kind() != T![===] {
```

or

```suggestion
    if expression.operator_token().ok()?.kind() != JsSyntaxKing::EQ3 {
```